HospitalSystem.Core/Scheduling/Appointments/AppointmentService.cs

```diff
@@ -0,0 +1,212 @@
+using MongoDB.Driver;
+using MongoDB.Bson;
+using HospitalSystem.Core.Utils;
+using HospitalSystem.Core.Rooms;
+
+namespace HospitalSystem.Core;
+
+public class AppointmentService
+{
+    private IAppointmentRepository _appointmentRepo;
+    private RoomService _roomService;
+    private DoctorService _doctorService;
+    private PatientService _patientService;
+
+    public AppointmentService(IAppointmentRepository appointmentRepo, RoomService roomService, DoctorService doctorService, PatientService patientService)
+    {
+        _appointmentRepo = appointmentRepo;
+        _roomService = roomService;
+        _doctorService = doctorService;
+        _patientService = patientService;
+    }
+
+    public void UpsertCheckup(Checkup newCheckup)
+    {
+        _appointmentRepo.UpsertCheckup(newCheckup);
+    }
+    public void UpsertOperation(Operation newOperation)
+    {
+        _appointmentRepo.UpsertOperation(newOperation);
+    }
+
+    public void DeleteCheckup(Checkup checkup)
+    {
+        _appointmentRepo.DeleteCheckup(checkup);
+    }
+
+    public void DeleteOperation(Operation operation)
+    {
+        _appointmentRepo.DeleteOperation(operation);
+    }
+
+    public void FinishCheckup(Checkup checkup)
+    {
+        checkup.Done = true;
+        _appointmentRepo.UpsertCheckup(checkup);
+    }
+
+     public IMongoCollection<Checkup> GetCheckups()
+    {
+        return _appointmentRepo.GetCheckups();
+    }
+
+    public IMongoCollection<Operation> GetOperations()
+    {
+        return _appointmentRepo.GetOperations();
+    }
+    public List<Checkup> GetCheckupsByPatient(ObjectId id)
+    {
+        var checkups = _appointmentRepo.GetCheckups();
+        List<Checkup> patientCheckups = checkups.Find(appointment => appointment.Patient.Id == id).ToList();
+        return patientCheckups;
+    }
+    
+    public List<Checkup> SearchPastCheckups(ObjectId patientId, string anamnesisKeyword)
+    {
+        var checkups = _appointmentRepo.GetCheckups();
+        //might not be the best way to indent
+        List<Checkup> filteredCheckups = 
+            (from checkup in checkups.AsQueryable().ToList()  // TODO: inefficient, but bug fix
+            where checkup.Anamnesis.ToLower().Contains(anamnesisKeyword.ToLower())
+            && checkup.DateRange.HasPassed() 
+            && checkup.Patient.Id == patientId
+            select checkup).ToList();
+        return filteredCheckups;
+    }
+
+    public List<Checkup> GetFutureCheckupsByPatient(ObjectId id)
+    {
+        var checkups = _appointmentRepo.GetCheckups();
+        List<Checkup> patientCheckups = 
+            (from checkup in checkups.AsQueryable().ToList()  // TODO: inefficient, but bug fix
+            where checkup.DateRange.IsFuture() && checkup.Patient.Id == id
+            select checkup).ToList();
+        return patientCheckups;
+    }
```

Comment:
Might want to put this in repo. Also, maybe dont use hasPassed and isFuture  (can't use them if querying the db directly)

Reply:
Same applies to GetFutureOpsByPat